datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1362,17 +1351,10 @@ impl DistributionContext {
 
     fn update_children(mut self) -> Result<Self> {
         for child_context in self.children_nodes.iter_mut() {
-            child_context.distribution_connection = match child_context.plan.as_any() {
-                plan_any if plan_any.is::<RepartitionExec>() => matches!(
-                    plan_any
-                        .downcast_ref::<RepartitionExec>()
-                        .unwrap()
-                        .partitioning(),
-                    Partitioning::RoundRobinBatch(_) | Partitioning::Hash(_, _)
-                ),
```

Comment:
Unnecessary matching. For all supported partitioning in `RepartitionExec`, `distribution_connection` should be true.

Reply:
And changed to use utility functions like `is_repartition`.